airflow/providers/microsoft/azure/secrets/azure_key_vault.py

```diff
@@ -0,0 +1,120 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from typing import Optional
+
+from azure.core.exceptions import ResourceNotFoundError
+from azure.identity import DefaultAzureCredential
+from azure.keyvault.secrets import SecretClient
+from cached_property import cached_property
+
+from airflow.secrets import BaseSecretsBackend
+from airflow.utils.log.logging_mixin import LoggingMixin
+
+
+class AzureKeyVaultBackend(BaseSecretsBackend, LoggingMixin):
+    """
+    Retrieves Airflow Connections or Variables from Azure Key Vault secrets.
+
+    The Azure Key Vault can be configred as a secrets backend in the ``airflow.cfg``:
+
+    .. code-block:: ini
+
+        [secrets]
+        backend = airflow.providers.microsoft.azure.secrets.azure_key_vault.AzureKeyVaultBackend
+        backend_kwargs = {"vault_url": "<azure_key_vault_uri>"}
+
+    For example, if the secrets prefix is ``airflow-connections-smtp-default``, this would be accessible
+    if you provide ``{"connections_prefix": "airflow-connections"}`` and request conn_id ``smtp-default``.
+    And if variables prefix is ``airflow-variables-hello``, this would be accessible
+    if you provide ``{"variables_prefix": "airflow-variables"}`` and request variable key ``hello``.
+
+    :param vault_url: The URL of an Azure Key Vault to use
+    :type vault_url: str
+    :param connections_prefix: Specifies the prefix of the secret to read to get Connections
+    :type connections_prefix: str
+    :param variables_prefix: Specifies the prefix of the secret to read to get Variables
+    :type variables_prefix: str
+    :param sep: separator used to concatenate secret_prefix and secret_id. Default: "-"
+    :type sep: str
+    """
+
+    def __init__(self, vault_url: str = '', connections_prefix: str = 'airflow-connections',
+                 variables_prefix: str = 'airflow-variables', sep: str = '-', **kwargs):
+        super().__init__()
+        self.vault_url = vault_url
```

Comment:
I think we should follow the pattern of the other secrets backend have `connections_prefix` and `variables_prefix` as the first 2 kwargs and move `vault_url` after those two.